x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/steps/details/reindex_details_step.tsx

```diff
@@ -0,0 +1,260 @@
+                  defaultMessage="The reindex operation allows transforming an index into a new, compatible one. It will copy all of the existing documents into a new index and remove the old one. Depending on size and resources, reindexing may take extended time and your data will be in a read-only state until the job has completed."
+                />
+              </p>
+            </Fragment>
+          )}
+          {!reindexState.meta.isReadonly && (
+            <Fragment>
+              <p>
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.index.flyout.detailsStep.notCompatibleIndexText"
+                  defaultMessage="This index was created in ES 7.x and it is not compatible with the next major version. You must address this issue before upgrading."
+                />
+              </p>
+              <ul>
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.index.flyout.detailsStep.reindexText"
+                  tagName="li"
+                  defaultMessage="The reindex operation allows transforming an index into a new, compatible one. It will copy all of the existing documents into a new index and remove the old one. Depending on size and resources, reindexing may take extended time and your data will be in a read-only state until the job has completed."
+                />
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.index.flyout.detailsStep.readOnlyText"
+                  tagName="li"
+                  defaultMessage="Alternatively, old indices can maintain compatibility with the next major version if they are turned into read-only mode. If you no longer need to update documents in this index (or add new ones), you might want to convert it to a read-only index."
+                />
+              </ul>
+            </Fragment>
+          )}
+          <p>
+            <FormattedMessage
+              id="xpack.upgradeAssistant.index.flyout.detailsStep.deleteText"
+              defaultMessage="If you no longer need this data, you can also proceed by deleting this index. {indexManagementLinkHtml}"
+              values={{
+                indexManagementLinkHtml: (
+                  <EuiLink
+                    href={`${http.basePath.prepend(
+                      `/app/management/data/index_management/indices/index_details?indexName=${indexName}`
+                    )}`}
+                  >
+                    <FormattedMessage
+                      id="xpack.upgradeAssistant.index.flyout.detailsStep.indexMgmtLink"
+                      defaultMessage="Go to index management"
+                    />
+                  </EuiLink>
+                ),
+              }}
+            />
+          </p>
+        </EuiText>
+        <EuiSpacer />
+      </EuiFlyoutBody>
+      <EuiFlyoutFooter>
+        <EuiFlexGroup justifyContent="spaceBetween">
```

Comment:
There is one requirement from the original issue regarding indices >1GB from the original issue that does not seem to be captured here:

> Offer to make user owned 7.x index read-only as the primary action if the 7.x index is >1GB, reindex as secondary action

and

> Offer to reindex a user owned 7.x index if the 7.x index is <= 1GB as the primary action, read-only as secondary action

But I think we should do this in a follow up PR.